### PR TITLE
Ben branch

### PR DESCRIPTION
**Pull request by Benflop**

### CODE REVIEW

File: src/main/java/DirectoryServiceAPI.java [hunk 1]
```diff
@@ -5,13 +5,16 @@
 import okhttp3.*;
 
 import java.io.IOException;
+import java.util.Scanner;
 
 public class DirectoryServiceAPI {
     private static DirectoryServiceAPI sharedInstance = null;
+    private String userToken;
```
Comment (clarencecodes) [inline src/main/java/DirectoryServiceAPI.java:12]:
```suggestion
```

File: src/main/java/DirectoryServiceAPI.java [hunk 2]
```diff
@@ -51,6 +54,14 @@ public Boolean login(String username, String password) throws IOException {
...
                 .getAsString();
 
+        RSAPUBLICKEY = jsonObject
+                .get("data")
+                .getAsJsonObject()
+                .get("user")
+                .getAsJsonObject()
+                .get("rsaPublicKey")
+                .getAsString();
+
```
Comment (clarencecodes) [inline src/main/java/DirectoryServiceAPI.java:64]:
```suggestion
```

We do not need to store the RSA public key from the server, as the RSA public and private keys will be generated on device.

File: src/main/java/DirectoryServiceAPI.java [hunk 3]
```diff
@@ -5,13 +5,16 @@
...
+import java.util.Scanner;
 
 public class DirectoryServiceAPI {
     private static DirectoryServiceAPI sharedInstance = null;
+    private String userToken;
 
 //    private String BASE_URL = "http://localhost:3000/api/";
     private String BASE_URL = "https://chat-directory-service.herokuapp.com/api/";
     private String TOKEN;
+    private String RSAPUBLICKEY;
```
Comment (clarencecodes) [inline src/main/java/DirectoryServiceAPI.java:17]:
```suggestion
```

File: src/main/java/DirectoryServiceAPI.java [hunk 4]
```diff
@@ -92,12 +103,14 @@ public String getIPAddressFromUsername(String username) throws IOException {
     public void UpdateIPAddress(String clientUsername, String ipAddress) throws IOException {
         String endpoint = BASE_URL + "address/update-ip-address";
         OkHttpClient client = new OkHttpClient();
-        String json = String.format("{\"username\": \"%s\", \"ipAddress\": \"%s\"}", clientUsername, ipAddress);
+
+        String json = String.format("{\"username\": \"%s\", \"ipAddress\": \"%s\", \"rsaPublicKey\": \"%s\"}", clientUsername, ipAddress, RSAPUBLICKEY);
```
Comment (clarencecodes) [inline src/main/java/DirectoryServiceAPI.java:107]:
```suggestion

        // TODO: remove this hardcoded value and replace it with the RSA public key from the end device
        String tempRsaPublicKey = "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA0JbajoEHlV1OqNLQeo6IzjaoOQaVPJn1UnBuuFK889vjFB6MCY2QH5gjQvd39D3lCL1TIdQac6RG19C5sx9rGzKwUsabA4ZljeMT5Nj2nRJnKvVaKc3Hj2xlxbbyK6nYsiXjdUfuQmaCWa5x+2yjzYYpmH5KFESfMTRdlfeHnP3T8psDjICdTRet949MadFrdpq1izAo88loMCJtPIAQWlNIvUGvE3+B/ysccOaJJzcPdHaTHm3+SSPMTMybJsgaAz8wezEXBJs2RwrMdX+CLG8sMZWq8InTObG6w638K/pkzMht/FExXw5KQwR9n8BdQ4GMYF1ykl5uqUS0yJxv6QIDAQAB";
        String json = String.format("{\"username\": \"%s\", \"ipAddress\": \"%s\", \"rsaPublicKey\": \"%s\"}", clientUsername, ipAddress, tempRsaPublicKey);
```

**Comment by Benflop** [comment]
Committed suggested changes.

**Pull request merged**